### DISCUSSION TITLE
Replacing photos with better quality photos

### DISCUSSION THREAD

**Discussion by dror3go** [photoprism/photoprism#2352] [Q&A] [answered]
Side from my digital photos and videos, I have several thousands of old scanned photos.
Those are already indexed, face-tagged, and sometimes grouped into albums.

I'm about to digitize at least some of them from film, which should result in higher quality photos.

How should I go about replacing the existing files with the new ones? Is there a supported way to do this? If no, should I 1) add the files next to the existing files 2) index, face-tag, add to existing albums 4) delete old files
Something else?

**Comment by graciousgrey** [comment, marked as answer]
I assume you talk about JPG files.

We never had this use case before. But you can try the following:

1)  Name the new files in a way that they will get stacked with the already existing ones, e.g. if the old one has the path holiday/IMG123.jpg the new one should be stacked if you add it as holiday/IMG123 (1).jpg (in case you have stacking by sequential names) enabled.

2) Set the new file  as primary file of the stack 

3) Delete the old file using the delete button from the files tab

4) Do a complete rescan of the folder

This way the new photo should keep most metadata you added manually (labels, description, dates). Also the photo stays in the albums it is part of.
Metadata that was read from the files metadata and not updated manually will be exchanged with the metadata from the new file. 
Faces are also associated with the file and you will need to re-do them.

**Reply by dror3go** [reply]
Thanks @graciousgrey. I'm aware that this an unusual use case, and I'm not expecting any dedicated new features to support such a thing. Just trying to understand the options available for me.

So all in all your solution sounds pretty good.
I just want to make sure two things about what you wrote:
1. Would there be a difference if I'll delete the lower-resolution files from outside of PhotoPrism? I guess it'll be easier for me to delete the files from the command line and not from the UI itself
2. New files being recognized as part of a stack of an existing files are automatically being added to all albums where the older files are?

**Reply by graciousgrey** [reply]
1. I did not test it with deleting from the filesystem. So I would recommend to test this with  a few files first.
2. In PhotoPrism "photos" are added to albums. A photo can consist of multiple files. If the non-primary file is deleted, the photo still exists with the second file and should still be part of the albums it was part of before.

**Reply by dror3go** [reply]
Amazing. Guess I have work to do.
Thanks a lot!

**Reply by alve89** [reply]
@dror3go May I ask you how you did this step-by-step? If I import a new version of a JPEG file, either with an incremented original name `IMG_8638 (1).JPG` or with an incremented new file name `20240512_132037_D3755E7E.jpg` (which is the automatic during initial import renamed name of `IMG_8638.JPG`), both files are added as new photos and get not stacked together with the old one.

The setting for incremented names is enabled:
![Bildschirmfoto 2024-05-20 um 23 20 23](https://github.com/photoprism/photoprism/assets/11487054/ff8620a1-cd60-4ec7-bc52-900010eecc6e)

I have the exact same issue as you.

@graciousgrey Or is there another way of importing an updated version of RAW + (darktable) XMP and re-create the primary JPEG of the photo?

**Reply by graciousgrey** [reply]
@alve89 If you want to add a new jpeg version that should be stacked with the file 2024/05/20240512_132037_D3755E7E.jpg you would need to add the new jpeg with a sequential name to the same location within the originals folder. In your case you would need to add a file 20240512_132037_D3755E7E (1).jpg to originals/2024/05/.

**Comment by dror3go** [comment]
@alve89 I chose not to keep the old versions (low quality scanning), so I'm not using stacks for this.
I "simple" replaced the old files with the new ones, so for example `/1970/11/25/IMG_19701125_a.jpg` is still the same filename, but the file itself is the new file.
I indexed the library once again after replacing the files, and PhotoPrism picked that up.

Keep in mind that while this solution worked for me - it's not officially supported as far as I understand, so bare in mind it might not be the right solution for you.
